src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs

```diff
@@ -39,7 +39,18 @@ public override int Read(byte[] buffer, int offset, int count)
 
         public override unsafe Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            return _httpContext.ReadAsync(buffer, offset, count, cancellationToken);
+            if (buffer == null)
```

Comment:
Can we just let the Memory constructor handle this?

Reply:
Doesn't throw with a null buffer.

Reply:
Then maybe it shouldn't.